src/EventStore.Core/Services/Monitoring/Utils/PerfCounterHelper.cs

```diff
@@ -14,8 +15,8 @@ internal class PerfCounterHelper : IDisposable {
 		public PerfCounterHelper(ILogger log) {
 			_log = log;
 
-			_totalCpuCounter = CreatePerfCounter("Processor", "% Processor Time", "_Total");
-			_totalMemCounter = CreatePerfCounter("Memory", "Available Bytes");
+			_totalCpuCounter = Runtime.IsWindows ? CreatePerfCounter("Processor", "% Processor Time", "_Total") : null;
+			_totalMemCounter = Runtime.IsWindows ? CreatePerfCounter("Memory", "Available Bytes") : null;
```

Comment:
If we move this check inside CreatePerfCounter then we only need to add a guard clause at the beginning (outside the try block) and if any subsequent counters ever get added, we won't forget that the perfcounters only exist on windows